ocpmodels/trainers/forces_trainer.py

```diff
@@ -472,22 +472,60 @@ def _compute_loss(self, out, batch_list):
-            force_mult = self.config["optim"].get("force_coefficient", 30)
-            if self.config["task"].get("train_on_free_atoms", False):
-                fixed = torch.cat(
-                    [batch.fixed.to(self.device) for batch in batch_list]
+            tag_specific_weights = self.config["task"].get(
+                "tag_specific_weights", []
+            )
+            if tag_specific_weights != []:
+                # handle tag specific weights as introduced in forcenet
+                assert len(tag_specific_weights) == 3
+
+                batch_tags = torch.cat(
+                    [
+                        batch.tags.float().to(self.device)
+                        for batch in batch_list
+                    ],
+                    dim=0,
                 )
-                mask = fixed == 0
-                loss.append(
-                    force_mult
-                    * self.criterion(out["forces"][mask], force_target[mask])
+                weight = torch.zeros_like(batch_tags)
+                weight[batch_tags == 0] = tag_specific_weights[0]
+                weight[batch_tags == 1] = tag_specific_weights[1]
+                weight[batch_tags == 2] = tag_specific_weights[2]
+
+                loss_force_list = torch.abs(out["forces"] - force_target)
+                train_loss_force_unnormalized = torch.sum(
+                    loss_force_list * weight.view(-1, 1)
                 )
-            else:
-                loss.append(
-                    force_mult * self.criterion(out["forces"], force_target)
+                train_loss_force_normalizer = 3.0 * weight.sum()
+
+                # add up normalizer to obtain global normalizer
+                distutils.all_reduce(train_loss_force_normalizer)
+
+                # perform loss normalization before backprop
+                train_loss_force_normalized = train_loss_force_unnormalized * (
+                    distutils.get_world_size() / train_loss_force_normalizer
                 )
+                loss.append(train_loss_force_normalized)
 
+            else:
+                # Force coefficient = 30 has been working well for us.
+                force_mult = self.config["optim"].get("force_coefficient", 30)
+                if self.config["task"].get("train_on_free_atoms", False):
+                    fixed = torch.cat(
+                        [batch.fixed.to(self.device) for batch in batch_list]
+                    )
+                    mask = fixed == 0
+                    loss.append(
+                        force_mult
+                        * self.criterion(
+                            out["forces"][mask], force_target[mask]
+                        )
+                    )
+                else:
+                    loss.append(
+                        force_mult
+                        * self.criterion(out["forces"], force_target)
+                    )
+                print(distutils.get_rank(), loss[-1])
```

Comment:
we can probably remove this print statement?

Reply:
Removed, thanks.